pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionFactory.java

```diff
@@ -227,14 +238,40 @@ public static AggregationFunction getAggregationFunction(FunctionContext functio
             }
           }
           case ARRAYAGG: {
-            Preconditions.checkArgument(numArguments == 2,
-                "ARRAY_AGG expects 2 arguments, got: %s. The function can be used as "
-                    + "arrayAgg(dataColumn, 'dataType')", numArguments);
+            Preconditions.checkArgument(numArguments >= 2,
+                "ARRAY_AGG expects 2 or 3 arguments, got: %s. The function can be used as "
+                    + "arrayAgg(dataColumn, 'dataType', ['isDistinct'])", numArguments);
```

Comment:
we should probably consider how to parse this. b/c the proper way to express distinct array agg is
```
SELECT key, ARRAY_AGG(DISTINCT col) 
FROM tbl 
GROUP BY key
```
given that our functions are not standard it is fine we do it now but good to note this

Reply:
also see: https://www.postgresql.org/docs/16/functions-aggregate.html#FUNCTIONS-AGGREGATE

Reply:
agreed, I was also thinking about this, however it's not parsed in v1 query engine.